py4cast/datasets/base.py

```diff
@@ -56,16 +56,20 @@ class NamedTensor(TensorWrapper):
     SPATIAL_DIM_NAMES = ("lat", "lon", "ngrid")
 
     def __init__(
-        self, tensor: torch.Tensor, names: List[str], feature_names: List[str]
+        self,
+        tensor: torch.Tensor,
+        names: List[str],
+        feature_names: List[str],
+        feature_dim_name: str = "features",
     ):
         if len(tensor.shape) != len(names):
             raise ValueError(
                 f"Number of names ({len(names)}) must match number of dimensions ({len(tensor.shape)})"
             )
-        if tensor.shape[-1] != len(feature_names):
+        if tensor.shape[names.index(feature_dim_name)] != len(feature_names):
             raise ValueError(
                 f"Number of feature names ({len(feature_names)}:{feature_names}) must match"
-                f"number of features ({tensor.shape[-1]})"
+                f"number of features ({tensor.shape[names.index(feature_dim_name)]})"
```

Comment:
could you make the exception message more explicit ?

Reply:
It looks like this :

```python
    _ = NamedTensor(
        torch.rand(3, 256, 256, 4),
        names=["batch", "lat", "lon", "features"],
        feature_names=[f"feature_{i}" for i in range(3)],
    )
```

which raises:
```python
ValueError: Number of feature names (3:['feature_0', 'feature_1', 'feature_2']) must matchnumber of features (4)
```
What do you have in mind  to make it more explicit ?

Reply:
What about this ? :

```python
ValueError: Number of feature names (3:['feature_0', 'feature_1', 'feature_2']) must matchnumber of features (4) in the supplied tensor
```